scRNA-seq/setup/glioblastoma/make_count_matrix.R

```diff
@@ -0,0 +1,114 @@
+# in one gene matrix tsv file.
+
+# Options:
+# "-d" - Directory of where individual samples' salmon folders are located.(Optional)
+# "-o" - Outfile for count matrix
+# "-r" - Outfile for tximport RDS matrix
+# "-p" - Outfile for plot 
+# "-m" - Percent mapped reads (reported as a decimal) cutoff for filtering
+#        samples. Default is 0.5.
+#
+# Command line example:
+#
+# Rscript scripts/make_count_matrix.R \
+# -d data/salmon_quants \
+# -o data/counts.tsv \
+# -r data/txi.RDS \
+# -p data/mapping_plot.png \
+# -m 0.5 
+
+#-------------------------- Get necessary packages-----------------------------#
+# Attach needed libraries
+library(optparse)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+#--------------------------------Set up options--------------------------------#
+option_list <- list(
+    make_option(c("-d", "--dir"), type = "character", default = getwd(),
+                help = "Directory where salmon quantification folders are located"),
+    make_option(c("-t", "--tx2gene"), type = "character",
+                help = "tx2gene table (tsv) for tximport"),
+    make_option(c("-o", "--outfile"), type = "character",
+                help = "Output file counts matrix (tsv)"),
+    make_option(c("-r", "--rds"), type = "character",
+                help = "Output file for tximport RDS object",
+                default = NULL),
+    make_option(c("-p", "--plot"), type = "character",
+                help = "Output file for mapping rate plot (png)",
+                default = NULL),  
+    make_option(c("-m", "--mapped"), type = "numeric",
+                default = "0.5", 
+                help = "Cutoff for what percent mapped_reads samples should have. 
+                Any samples with less than the cutoff will be removed.")
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+#------------------------------Import Salmon reads-----------------------------#
+# Get quant files
+quant_files <- list.files(opt$dir, recursive = TRUE, full.names = TRUE,
+                          pattern = "quant.sf")
+
+# Get sample names
+sample_names <- stringr::word(quant_files, -2, sep = "/")
+
+# Get transcript IDs
+transcripts <- read.table(quant_files[[1]], header = TRUE,
+                          colClasses = c("character", rep("NULL", 4)))
+
+# Get rid of transcript version numbers because mapIDs will not recognize them
+transcripts.wout.ver <- gsub("\\.[0-9]*$", "", transcripts$Name)
+
+# Read tx2gene file
+tx2gene_df <- readr::read_tsv(opt$tx2gene)
+
+# Do the thing
+txi <- tximport::tximport(quant_files, 
+                          type = "salmon", 
+                          tx2gene = tx2gene_df,
+                          countsFromAbundance = "no",
+                          ignoreTxVersion = TRUE)
+
+# Save to RDS file temporarily
+if (is.character(opt$rds)){
+  readr::write_rds(txi, opt$rds)
+}
+
+# counts_df <- readRDS("tximport_obj.RDS")
```

Comment:
What's this about?

Reply:
You tell me! I should remove it, but I think the deal was that you were using it for checkpointing in the original script.

Reply:
Turns out, it is removed in my local branch....